CenterForOpenScience/osf.io
# [OSF-8056] Fix for addon asset loading

## Purpose

One staging and production the osf will try to grab js assets for directories which don't exist causing a key error. This happens on addons because default addons, like osfstorage are listed on the addons page, but don't actually need any js assets because they are always enabled and configured the same way. This fix ensures we never try to load non-existent  assets for default addons. 

Currently osfstorage is the only default addon.

## Changes

Cleans up the code that loads js assets into the mako file by adding comments and unittests.

## QA Notes

Check to make sure there are no gaps in the lower level panels, the ones that say "connect account" on them.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/OSF-8056
